### PR TITLE
add new api docs

### PR DESCRIPTION
**Pull request by shaunlebron**
__[Preview the new markdown API doc here](https://github.com/shaunlebron/paperprograms/blob/api/docs/api.md)__

This PR also redirects the [original api page](https://paperprograms.org/api.html) to it.

### CODE REVIEW

File: www/api.html [hunk 1]
```diff
@@ -1,101 +1,4 @@
...
-      <li><code>whiskerLength</code>: as fraction of the side (default 0.7);</li>
-      <li><code>requiredData</code>: array of data fields that must be present in the other paper;</li>
-      <li><code>paperNumber</code>: paper number to do this for (default is own paper number).</li>
-    </ul>
-  </p>
-</body>
+<script>
+  // Redirect to new API documentation page:
+  window.location.replace("https://github.com/janpaul123/paperprograms/blob/master/docs/api.md");
+</script>
```
Comment (janpaul123) [inline www/api.html]: Maybe just delete this page entirely and link directly to `https://github.com/janpaul123/paperprograms/blob/master/docs/api.md` from the editor page?

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+Each marker has the following properties:
+
+```js
+{
+  position: {x, y},  // global position
+  color: [r, g, b]
+  colorName // "red", "green", "blue" or "black"
+
+  // these properties will be null if the marker is not on a paper
+  paperId,
```
Comment (janpaul123) [inline docs/api.md]: Only noticing this now, but should we call this `paperNumber` for consistency? (This is outside the scope of this PR though, and no biggie anyway)

Comment (shaunlebron) [reply docs/api.md]: ah, I pulled this from https://github.com/janpaul123/paperprograms/pull/43#discussion_r182001387, but it looks like it was not implemented.  fixed the name, and tagged as wip.

File: www/api.html (hunk 1 above)
Comment (janpaul123) [reply www/api.html]: Sure, though I doubt that many people will have this bookmarked since it's right there on the editor page. Let's at least then also update the link though, no need to link to an old page.

Comment (shaunlebron) [reply www/api.html]: yeah, maybe I'm on the only one who bookmarked it.  just updated the editor link 👍

Comment (janpaul123) [reply www/api.html]: Thanks!
